hoomd/trigger.py

```diff
@@ -2,15 +2,45 @@
 # This file is part of the HOOMD-blue project, released under the BSD 3-Clause
 # License.
 
+""" :py:class:`hoomd.trigger` enables users to design time points when and while
+    :py:class:`hoomd.Operations` operates. The composite of triggers takes 
+    time steps and returns `True` or `False`. The operation will perform when 
+    Triger returns `True`.
+
+"""
+
 from hoomd import _hoomd
 from inspect import isclass
 
 
-class Trigger(_hoomd.Trigger):
+class Trigger(_hoomd.Trigger):  # noqa: D101
```

Comment:
We should document this class.
Something like:
```
"""Base class trigger.

Provides methods common to all triggers.

Attention:
    Users should instantiate the subclasses, using `Trigger` directly
    will result in an error.
"""
```

We also need to document the `compute` and `__call__` methods. I need to do some checking and see if I can figure out how to do that with the way we set these classes up in sphinx.